lib/png/png.go

```diff
@@ -0,0 +1,103 @@
+package png
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	_ "embed"
+
+	"github.com/playwright-community/playwright-go"
+)
+
+func InitPlaywright() (*playwright.Playwright, playwright.Browser, error) {
+	// check if playwright driver/browsers are installed and up to date
+	// https://github.com/playwright-community/playwright-go/blob/8e8f670b5fa7ba5365ae4bfc123fea4aac359763/run.go#L64.
+	driver, err := playwright.NewDriver(&playwright.RunOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	if _, err := os.Stat(driver.DriverBinaryLocation); errors.Is(err, os.ErrNotExist) {
+		err = playwright.Install()
+		if err != nil {
+			return nil, nil, err
+		}
+	} else if err == nil {
+		cmd := exec.Command(driver.DriverBinaryLocation, "--version")
+		output, err := cmd.Output()
+		if err != nil || !bytes.Contains(output, []byte(driver.Version)) {
+			err = playwright.Install()
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+	}
+
+	pw, err := playwright.Run()
+	if err != nil {
+		return nil, nil, err
+	}
+	browser, err := pw.Chromium.Launch()
+	if err != nil {
+		return nil, nil, err
+	}
+	return pw, browser, nil
+}
+
+//go:embed generate_png.js
+var genPNGScript string
+
+func ExportPNG(browser playwright.Browser, svg []byte) (outputImage []byte, err error) {
+	var page playwright.Page
+	defer func() error {
+		err = page.Close()
+		if err != nil {
+			return err
+		}
+		return nil
+	}()
+
+	if browser == nil {
+		return nil, fmt.Errorf("browser is not initialized for png export")
+	}
+	page, err = browser.NewPage()
```

Comment:
is this the right usage? do we need to keep opening and closing new pages?

from https://playwright.dev/docs/api/class-browser#browser-new-page


> This is a convenience API that should only be used for the single-page scenarios and short snippets. Production code and testing frameworks should explicitly create [browser.newContext([options])](https://playwright.dev/docs/api/class-browser#browser-new-context) followed by the [browserContext.newPage()](https://playwright.dev/docs/api/class-browsercontext#browser-context-new-page) to control their exact life times.

Reply:
Interesting. In this case I'm gonna create a new object to store all the various parts of playwright instead of passing them all around.

Reply:
There's a little bit of trickiness here. I can only close browserContext properly when we don't use watcher. If we command+c out of watcher and the browserContext.close() function throws an error saying the browser is already closed, not sure why. Maybe the signal interrupt just shuts everything down ASAP?

Reply:
it could always be in a "closed" state, we should be able to handle that (ignore)

Reply:
The weird thing is Browser.IsConnected() returns true when browserContext.close() is called. Based on what the creator says here though https://github.com/playwright-community/playwright-go/issues/247#issuecomment-1065885796, it seems like Browser.close() should be enough as it closes everything underneath it.